torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -68,6 +68,49 @@ c10::optional<at::ScalarType> PromoteIntegralType(
                                                                    : opt_dtype;
 }
 
+bool IsTypeWithLargerRangeThanLong(torch::ScalarType dtype) {
+  return dtype == at::ScalarType::BFloat16 || dtype == at::ScalarType::Float ||
+         dtype == at::ScalarType::Double;
+}
+
+// Return the upper limit for a given type. For floating point typesreturn
+// 2^mantissa to ensure that every value is representable.
+int64_t GetIntegerUpperLimitForType(torch::ScalarType dtype) {
+  xla::PrimitiveType xla_type = TensorTypeToRawXlaType(dtype);
+  if (isFloatingType(dtype)) {
```

Comment:
I think here you can just do (sorry if I misled you):
```c++
switch (type) {
  case F16:
    return ...;
  case ...:
    return ...;
  default:
    return XlaHelpers::MinMaxValues(...);
}
```

Reply:
Ah, I see. I understand it wrong 😄